src/configured_yang_push.h

```diff
@@ -0,0 +1,104 @@
+/*
+ * @file configured_yang_push.h
+ * @author Jeremie Leska <jeremie.leska@6wind.com>
+ * @brief ietf-subscribed-notifications configured ietf-yang-push callbacks
+ * header
+ *
+ * @copyright
+ * Copyright 2023 6WIND S.A.
+ *
+ * This source code is licensed under BSD 3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://opensource.org/licenses/BSD-3-Clause
+ */
+
+#ifndef NP2SRV_CONFIGURED_YANG_PUSH_H_
+#define NP2SRV_CONFIGURED_YANG_PUSH_H_
+
+#include <stdint.h>
+
+#include <libyang/libyang.h>
+#include <sysrepo.h>
+
+#include "common.h"
+#include "netconf_subscribed_notifications.h"
+#include "receivers.h"
+#include "yang_push.h"
+
+struct np2srv_sub_ntf;
+
+struct cfg_yang_push_cb_arg {
+    /* reference to receivers */
+    struct csn_receiver_info recv_info;
+
+    struct cfg_yang_push_data *cyp_data;
+    uint32_t nc_sub_id;
+};
+
+struct cfg_yang_push_data {
+    /* parameters */
+    sr_datastore_t datastore;
+
+    char *selection_filter_ref;
+    struct lyd_node *datastore_subtree_filter;
+    char *datastore_xpath_filter;
+    int periodic;
+
+    union {
+        struct {
+            /* parameters */
+            uint32_t period_ms;
+            struct timespec anchor_time;
+
+            /* internal data */
+            timer_t update_timer;
+        };
+        struct {
+            /* parameters */
+            uint32_t dampening_period_ms;
+            int sync_on_start;
+            int excluded_change[YP_OP_OPERATION_COUNT];
+
+            /* internal data */
+            pthread_mutex_t notif_lock;
+            sr_data_t *change_ntf;
+            ATOMIC_T patch_id;
+            ATOMIC_T edit_id;
+            struct timespec last_notif;
+            timer_t damp_timer;
+            ATOMIC_T excluded_op_count; /* explicitly excluded changes */
+        };
+    };
+
+    /* internal data */
+    char *xpath;
+    struct cfg_yang_push_cb_arg cb_arg;
+    timer_t stop_timer;
+    enum configured_subscription_state state;
+};
+
+/* for documentation, see subscribed_notifications.h */
+int
```

Comment:
For function declarations we put the return type and function name on the same line.